packages/contracts/contracts/plugin/PluginSetupProcessor.sol

```diff
@@ -604,4 +617,55 @@ contract PluginSetupProcessor is DaoAuthorizable {
             });
         }
     }
+
+    /// @notice Checks the update parameters
+    /// @param _updateParams The parameters of the update.
+    /// @return oldVersion The old semantic version number being needed in the context for further usage.
+    function _checkUpdateValidity(PluginUpdateParams calldata _updateParams)
+        internal
+        view
+        returns (uint16[3] memory oldVersion)
+    {
+        // These revert if the current or new plugin setup are not part of the plugin setup repo
+        (oldVersion, , ) = _updateParams.pluginSetupRepo.getVersionByPluginSetup(
+            _updateParams.currentPluginSetup
+        );
+        (uint16[3] memory newVersion, , ) = _updateParams.pluginSetupRepo.getVersionByPluginSetup(
+            _updateParams.newPluginSetup
+        );
+
+        // Assert that the version bump valid
+        if (!isValidUpdate(oldVersion, newVersion)) {
+            revert UpdateInvalid({currentVersion: oldVersion, invalidNextVersion: newVersion});
+        }
+    }
+
+    /// @notice Checks if an update is valid by the version numbers.
+    /// @param _oldVersion The old semantic version number.
+    /// @param _newVersion The new semantic version number.
+    /// @return bool Returns true if the bump is valid.
+    function isValidUpdate(uint16[3] memory _oldVersion, uint16[3] memory _newVersion)
+        public
+        pure
+        returns (bool)
+    {
+        bool hasIncreased;
+        uint256 i = 0;
+        while (i < 3) {
+            if (hasIncreased) {
+                if (_newVersion[i] != 0) {
+                    return false;
+                }
+            } else if (_newVersion[i] != _oldVersion[i]) {
+                if (_oldVersion[i] > _newVersion[i]) {
+                    return false;
+                }
+                hasIncreased = true;
```

Comment:
wouldn't returning when it has increased be enough? otherwise we force users to follow the update path step by step.
For example if I wanna update from 1.0.2 to 1.1.1 this isn't allowed.

Reply:
Resolved in DM and clarified further by splitting isValidBump into two functions (strict and loose mode).